### PR TITLE
Added FileRentention Utility to clean up files older than retention date

### PR DESCRIPTION
**Pull request by naishu**
"

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+    public void cleanup(String dir_path,int num_of_rentention_days) throws IOException {
+
+        Date Retention_date = DateUtils.addDays(new Date(), num_of_rentention_days);
+        long Retention_date_epoch = Retention_date.getTime() / 1000L;
+
+        try (Stream<Path> filePathStream = Files.walk(Paths.get(dir_path))) {
+            filePathStream.forEach(filePath -> {
+                if (Files.isRegularFile(filePath)) {
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:23]: can you add support for dir deletion?

File: druid-lookups/src/test/java/com/yahoo/maha/maha_druid_lookups/query/lookup/FileRetentionTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        FileRenention.cleanup(String.valueOf(tempDirWithPrefix),-10);
+
+        File f = new File(String.valueOf(tempDirWithPrefix));
+        pathnames = f.list();
+        Assert.assertEquals(pathnames.length, 2);
+        file3.deleteOnExit();
+        file4.deleteOnExit();
+
+    }
```
Comment (pranavbhole) [inline druid-lookups/src/test/java/com/yahoo/maha/maha_druid_lookups/query/lookup/FileRetentionTest.java:45]: empty lines?

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                    if(filePath.toFile().lastModified() <= Retention_date_epoch)
+                    {
+                        filePath.toFile().delete();
+                       LOG.info(filePath.toFile().getName() + "deleted");
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:27]: log full path

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.stream.Stream;
+
+public class FileRetention {
+    private static final Logger LOG = new Logger(FileRetention.class);
+
+
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:15]: java comment

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        Date Retention_date = DateUtils.addDays(new Date(), num_of_rentention_days);
+        long Retention_date_epoch = Retention_date.getTime() / 1000L;
+
+        try (Stream<Path> filePathStream = Files.walk(Paths.get(dir_path))) {
+            filePathStream.forEach(filePath -> {
+                if (Files.isRegularFile(filePath)) {
+                    if(filePath.toFile().lastModified() <= Retention_date_epoch)
+                    {
+                        filePath.toFile().delete();
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:26]: try catch wrapper,  for each dir to be deleted

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+                    }
+
+                }
+            });
+
+        }
+        catch (Exception e)
+        {
+            LOG.info("Exception while trying to clean up "+ e);
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:37]: log error

File: druid-lookups/src/test/java/com/yahoo/maha/maha_druid_lookups/query/lookup/FileRetentionTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        File file1 = File.createTempFile( "Test", "1", new File("target/testCleanup"));
+        File file2 = File.createTempFile( "Test", "2", new File("target/testCleanup"));
+        File file3 = File.createTempFile( "Test", "3", new File("target/testCleanup"));
+        File file4 = File.createTempFile( "Test", "4", new File("target/testCleanup"));
+
+
+        file1.setLastModified(Retention_date_epoch);
+        file2.setLastModified(Retention_date_epoch);
+
+        FileRenention.cleanup(("target/testCleanup"),-10);
```
Comment (pranavbhole) [inline druid-lookups/src/test/java/com/yahoo/maha/maha_druid_lookups/query/lookup/FileRetentionTest.java:36]: library function should not take minus

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                            filePath.toFile().delete();
+                            LOG.info(filePath.toFile().getName() + "at Path -" + filePath.toFile().getAbsolutePath() + "deleted");
+
+                        } catch (Exception e) {
+                            LOG.error("Exception while trying to delete file " + e);
+                        }
+
+
+                    }
+
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:42]: remove empty lines

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        Date Retention_date = DateUtils.addDays(new Date(), num_of_rentention_days * -1);
+        long Retention_date_epoch = Retention_date.getTime() / 1000L;
+
+        try (Stream<Path> filePathStream = Files.walk(Paths.get(dir_path))) {
+            filePathStream.forEach(filePath -> {
+                if (Files.isRegularFile(filePath)) {
+                    if (filePath.toFile().lastModified() <= Retention_date_epoch) {
+                        try {
+                            filePath.toFile().delete();
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:33]: force delete

Comment (naishu) [reply druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java]: done

File: druid-lookups/src/test/java/com/yahoo/maha/maha_druid_lookups/query/lookup/FileRetentionTest.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        FileRenention.cleanup(String.valueOf(tempDirWithPrefix),-10);
+
+        File f = new File(String.valueOf(tempDirWithPrefix));
+        pathnames = f.list();
+        Assert.assertEquals(pathnames.length, 2);
+        file3.deleteOnExit();
+        file4.deleteOnExit();
+
+    }
+
```
Comment (pranavbhole) [inline druid-lookups/src/test/java/com/yahoo/maha/maha_druid_lookups/query/lookup/FileRetentionTest.java:46]: empty lines again?

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java (hunk 6 above)
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:37]: log e as argument to error logger

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        try (Stream<Path> filePathStream = Files.walk(Paths.get(dir_path))) {
+            filePathStream.forEach(filePath -> {
+                if (Files.isRegularFile(filePath)) {
+                    if(filePath.toFile().lastModified() <= Retention_date_epoch)
+                    {
+                        try {
+                            FileDeleteStrategy.FORCE.delete(filePath.toFile());
+                            LOG.info("File " + filePath.toFile().getName() + "deleted at path " + filePath.toFile().getAbsolutePath());
+                        } catch (IOException e) {
+                            LOG.error("exception when trying to delete " + filePath.toFile().getName() + e);
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:36]: passed e as args

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+                        }
+
+                    }
+                }
+            });
+        }
+        catch (Exception e)
+        {
+            LOG.error("Exception while trying to clean up "+ e);
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/FileRetention.java:45]: same

**Review by pranavbhole — APPROVED** [review]

**Pull request merged**